plugin/manager.go

```diff
@@ -84,6 +82,78 @@ func (m *CLIManager) List(ctx context.Context) ([]string, error) {
 	return plugins, nil
 }
 
+// Install installs a plugin at filePath to the system.
+// It returns the new plugin metadata and a nil error if and only if
+// the installation succeeded.
+//
+// If plugin does not exist, directly install from filePath.
+//
+// If plugin already exists:
+//
+// If overwrite is not set, then the new plugin
+// version MUST be higher than the existing plugin version.
+//
+// If overwrite is set, version check is skipped. If existing
+// plugin is malfunctioning, it will be overwritten.
+func (m *CLIManager) Install(ctx context.Context, filePath string, overwrite bool) (*proto.GetMetadataResponse, *proto.GetMetadataResponse, error) {
```

Comment:
@shizhMSFT This is a good one. The problem is that we do not have a clearly stated format for a plugin publisher to follow in the spec. It is tracked in this issue created by @FeynmanZhou: https://github.com/notaryproject/specifications/issues/284

Considering extensibility, I have updated the input to `InstallOptions` and moved `FilePath` there. If we decide to support more formats in the future, we could add them there to avoid any breaking change.